File: questao_02/questao_02.go
package main

import (
	"fmt"
)

type Carro struct {
	marca    string
	modelo   string
	ano      int
	velocidade int
}

func (c *Carro) Acelerar(valor int) {
	c.velocidade += valor
	fmt.Printf("%s %s acelerou para %d km/h.\n", c.marca, c.modelo, c.velocidade)
}

func (c *Carro) Frear(valor int) {
	c.velocidade -= valor
	if c.velocidade < 0 {
		c.velocidade = 0
	}
	fmt.Printf("%s %s freou para %d km/h.\n", c.marca, c.modelo, c.velocidade)
}

func (c Carro) ExibirVelocidade() {
	fmt.Printf("Velocidade atual de %s %s: %d km/h\n", c.marca, c.modelo, c.velocidade)
}

func main() {
	carro1 := Carro{"Peugeot", "208", 2021}
	
	carro1.Acelerar(50)
	carro1.ExibirVelocidade()
	carro1.Frear(20)
	carro1.ExibirVelocidade()
	carro1.Frear(40)
	carro1.ExibirVelocidade()
}

File: questao_02/questao_02.java
class Carro {
    private String marca;
    private String modelo;
    @SuppressWarnings("unused")
    private int ano;
    private int velocidade;

    public Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    public void acelerar(int valor) {
        velocidade += valor;
        System.out.printf("%s %s acelerou para %d km/h.%n", marca, modelo, velocidade);
    }

    public void frear(int valor) {
        velocidade -= valor;
        if (velocidade < 0) {
            velocidade = 0;
        }
        System.out.printf("%s %s freou para %d km/h.%n", marca, modelo, velocidade);
    }

    public void exibirVelocidade() {
        System.out.printf("Velocidade atual de %s %s: %d km/h%n", marca, modelo, velocidade);
    }

    public static void main(String[] args) {
        Carro carro1 = new Carro("Peugeot", "208", 2021);
        
        carro1.acelerar(50);
        carro1.exibirVelocidade();
        carro1.frear(20);
        carro1.exibirVelocidade();
        carro1.frear(40);
        carro1.exibirVelocidade();
    }
}
